dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -230,79 +219,80 @@ public Response generateIntegrityData(@Context HttpServletRequest request, @Form
+                    IntegrityDataGenerationJob.JOB_GROUP_NAME)) {
+                return Response.status(HttpStatus.SC_PROCESSING).build();
             }
 
-            ProcessStatus integrityDataGeneratorStatus = (ProcessStatus) servletContext.getAttribute("integrityDataGenerationStatus");
-
-            if(UtilMethods.isSet( integrityDataGeneratorStatus )) {
-                switch (integrityDataGeneratorStatus) {
-                    case PROCESSING:
-                        return Response.status(HttpStatus.SC_PROCESSING).build();
-                    case FINISHED:
-                        StreamingOutput output = new StreamingOutput() {
-                            public void write(OutputStream output) throws IOException, WebApplicationException {
-                                InputStream is = Files.newInputStream(Paths.get(
-                                        ConfigUtils.getIntegrityPath() + File.separator
-                                                + requesterEndPoint.getId() + File.separator
-                                                + INTEGRITY_DATA_TO_CHECK_ZIP_FILE_NAME));
-
-                                byte[] buffer = new byte[1024];
-                                int bytesRead;
-                                //read from is to buffer
-                                while((bytesRead = is.read(buffer)) !=-1){
-                                    output.write(buffer, 0, bytesRead);
-                                }
-                                is.close();
-                                //flush OutputStream to write any buffered data to file
-                                output.flush();
-                                output.close();
-
-                            }
-                        };
-                        return Response.ok(output).build();
+            final Optional<IntegrityUtil.IntegrityDataExecutionMetadata> controlOpt =
+                    IntegrityUtil.getIntegrityDataControl(requesterEndpoint.getId());
+            if (!controlOpt.isPresent() || !requestId.equals(controlOpt.get().getRequestId())) {
+                return Response.status(HttpStatus.SC_UNAUTHORIZED).build();
+            }
 
-                    case CANCELED:
-                        return Response.status( HttpStatus.SC_RESET_CONTENT ).entity( servletContext.getAttribute( "integrityDataGenerationError" ) ).build();
+            if (controlOpt.get().getProcessStatus() == ProcessStatus.PROCESSING) {
+                return Response.status(HttpStatus.SC_PROCESSING).build();
+            } else if (controlOpt.get().getProcessStatus() == ProcessStatus.FINISHED &&
+                    IntegrityUtil.doesIntegrityDataFileExist(
+                            requesterEndpoint.getId(),
+                            IntegrityUtil.INTEGRITY_DATA_TO_CHECK_ZIP_FILENAME)) {
+                final StreamingOutput output = so -> {
+                    final InputStream is = Files.newInputStream(
```

Comment:
Codacy found an issue: [Avoid variables with short names like is](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)